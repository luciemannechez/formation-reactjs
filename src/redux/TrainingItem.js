// Composant de type Class
// Une propriété props est définie dans l'objet (accessible via this.props)
// Les propriétés de cet objet sont les attributs XML déclarés lorsque le composant est instancié
import React from 'react';
import PropTypes from 'prop-types';

export default class TrainingItem extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    console.log(this.props.name);
  }

  render() {
    return (
        <li>{this.props.name}</li>
    );
  }
}

TrainingItem.propTypes = {
  name: PropTypes.string.isRequired,
};
