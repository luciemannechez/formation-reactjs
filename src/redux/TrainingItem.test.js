import React from 'react';
import renderer from 'react-test-renderer';
import TrainingItem from './TrainingItem';

describe('TrainingItem.js', () => {
    it('should take a snapshot', () => {
        const tree = renderer.create(<TrainingItem name="Nicolas"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
