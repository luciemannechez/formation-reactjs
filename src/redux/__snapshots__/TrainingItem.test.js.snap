// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingItem.js should take a snapshot 1`] = `
<li>
  Nicolas
</li>
`;
